zhmcclient/_resource.py

```diff
@@ -174,3 +174,29 @@ def get_property(self, name):
                 raise
             self.pull_full_properties()
             return self._properties[name]
+
+    def __str__(self):
+        """
+        Convert a BaseResource object to a string.
+        Used for printing.
+
+        example:
+        <Cpc _full_properties=False, _image_activation_profiles=None,
+        _load_activation_profiles=None, _lpars=None,
+        _manager=<zhmcclient._cpc.CpcManager object at 0x7f439cab3510>,
+        _partitions=None, _properties={u'status': u'service-required',
+        u'object-uri': u'/api/cpcs/f1bc49af-f71a-3467-8def-3c186b5d9352',
+        u'name': u'P0000P01'}, _properties_timestamp=1472558081,
+        _reset_activation_profiles=None,
+        _uri=/api/cpcs/f1bc49af-f71a-3467-8def-3c186b5d9352>
+        """
+        sortedkeys = sorted(self.__dict__.keys())
+        info = ", ".join("%s=%s" % (k, getattr(self, k)) for k in sortedkeys)
+        return "<%s %s>" % (self.__class__.__name__, info)
+
+    def __repr__(self):
+        """
+        Convert a BaseResource object to astring.
+        Used for printing a list containing a BaseResource object.
```

Comment:
"Used for printing" does not really define the purpose. "Used as debug information" would be better, I think.

Also, this is a single resource. Not sure I understand why it says "a list containing a BaseResource object"?